### PR TITLE
feat: add tally/prefer-copy-chmod rule

### PR DESCRIPTION
**Pull request by tinovyatkin**
## Summary

- Add new `tally/prefer-copy-chmod` rule that detects `COPY` + `RUN chmod` pairs and suggests merging them into `COPY --chmod`, reducing image layers
- Handle COPY with existing `--chmod` by merging modes (e.g. `--chmod=644` + `chmod +x` → `--chmod=0755`) via new `shell.MergeChmodModes` utility
- Simplify `prefer-copy-heredoc` to pass symbolic chmod modes (`+x`, `u+rwx`) through to `COPY --chmod` directly instead of converting to octal — supported since Dockerfile 1.14
- Export `IsOctalMode`, `IsSymbolicMode`, `ApplySymbolicMode` from `internal/shell/` for reuse across rules
- Cross-rule integration test (`TestFixCrossRuleCopyChmodHeredoc`) verifies both COPY rules cooperate without fix conflicts

## Test plan

- [x] Unit tests: 42 cases covering detection, fixes, merge logic, edge cases
- [x] Integration lint test with realistic multi-stage fixture
- [x] Integration fix tests: basic, with `--chown`, directory destination
- [x] Cross-rule fix test: `prefer-copy-chmod` + `prefer-copy-heredoc` simultaneously
- [x] Real-world fixture (`TestFixRealWorld`) updated and passing
- [x] `go test ./...` — all packages pass
- [x] `make lint` — 0 issues
- [x] `make cpd` — clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a new tally rule "prefer-copy-chmod" (Style, Info) that auto-fixes COPY + RUN chmod into COPY --chmod (enabled by default).

* **Bug Fixes / Improvements**
  * Improved cross-rule auto-fix cooperation so multiple rules can fix the same file in one run.
  * Formatting/LSP now prefers setting executable permissions during COPY and preserves original chmod notation (octal/symbolic).

* **Documentation**
  * Added rule docs and updated supported-rules listings/examples.

* **Tests**
  * Added integration/unit tests and updated snapshots for detection, fixes, and cross-rule scenarios.

### CODE REVIEW

File: internal/rules/tally/prefer_copy_chmod.go [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+			if info := sem.StageInfo(stageIdx); info != nil {
+				shellVariant = info.ShellSetting.Variant
+			}
+		}
+
+		if !shellVariant.SupportsPOSIXShellAST() {
+			continue
+		}
+
+		violations = append(violations, r.checkStage(stage, shellVariant, input.File, sm, meta)...)
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/rules/tally/prefer_copy_chmod.go]: ✅ Addressed in 75690d3. Thanks @coderabbitai\!

File: internal/rules/tally/prefer_copy_chmod_test.go [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+			WantViolations: 1,
+		},
+		{
+			Name: "existing --chmod + redundant chmod",
+			Content: `FROM alpine
+COPY --chmod=777 script.sh /app/script.sh
+RUN chmod +x /app/script.sh
+`,
+			WantViolations: 1, // still flag — RUN is redundant
+		},
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/rules/tally/prefer_copy_chmod_test.go]: ✅ Addressed in eafaeb8. Thanks @coderabbitai\!
